src/core/ext/transport/chttp2/transport/chttp2_transport.cc

```diff
@@ -1179,6 +1179,17 @@ void grpc_chttp2_add_incoming_goaway(grpc_chttp2_transport* t,
   /* lie: use transient failure from the transport to indicate goaway has been
    * received */
   connectivity_state_set(t, GRPC_CHANNEL_TRANSIENT_FAILURE, "got_goaway");
+
+  /* Cancel streams with id larger than last_stream_id. */
```

Comment:
we don't want to do this yet. We want to ideally follow a double goaway mechanism where -
we first send a goaway to indicate that the server wants to shut down, and a second goaway after some time to indicate that no new requests will be served. If we do this right now, many RPCs are gonna start failing.